File: code/backend/crates/sea-weir-core/src/admin/system.rs
//! 系统域:首装向导(setup)、系统状态(status)、公告/关于/协议文案
//!
//! 契约见 doc/architecture/CONTRACTS.md。

// TODO(TDD): 定义本域的 Service 结构体与方法。
//   依赖以 `Arc<dyn XxxRepository>` 注入,便于用 mock 单测。
//
// 本域测试要点:
//   未初始化时 PostSetup 创建 root;已初始化后再次调用被拒;status 返回的开关集合与前端 store 期望一致

File: code/frontend/src/api/modules/system.ts
/** `system` 域 API。端点清单见 doc/architecture/CONTRACTS.md 附录 A。 */
import { request } from '@/api/client';
import type { SetupInfo, SystemStatus } from '@/types';

/** `GET /api/status`(公开):全站配置,status store 的唯一来源。 */
export function getStatus(): Promise<SystemStatus> {
  return request<SystemStatus>({ url: '/api/status', method: 'get' });
}

/** `GET /api/setup`(公开):首装状态。 */
export function getSetup(): Promise<SetupInfo> {
  return request<SetupInfo>({ url: '/api/setup', method: 'get' });
}

/** `POST /api/setup`(公开):未初始化时创建 root。已初始化后端拒绝。 */
export function postSetup(payload: {
  username: string;
  password: string;
}): Promise<{ id: number; username: string }> {
  return request<{ id: number; username: string }>({
    url: '/api/setup',
    method: 'post',
    data: payload,
  });
}

/** `GET /api/notice`(公开):站内公告文案(字符串,可含 Markdown/HTML)。 */
export function getNotice(): Promise<string> {
  return request<string>({ url: '/api/notice', method: 'get' });
}
